python/dgl/distributed/dist_context.py

```diff
@@ -44,7 +44,23 @@ def _init_rpc(ip_config, num_servers, max_queue_size, net_type, role, num_thread
 def initialize(ip_config, num_servers=1, num_workers=0,
                max_queue_size=MAX_QUEUE_SIZE, net_type='socket',
                num_worker_threads=1):
-    """Init rpc service
+    """Initialize DGL's distributed module
+
+    This function has to be invoked at the very beginning of the training script.
```

Comment:
It's better to say: Users cannot invoke any other distributed API before initializing.

Reply:
The term `distributed API` is still not accurate enough because it includes both DGL distributed API and *framework-specific* distributed API.  You can be more specific on this term (something like "You need to initialize the module before calling any DGL distributed API or framework-specific distributed API.").

Also would you care explaining the reason why this has to be called before `init_process_group`?  And how about MXNet or Tensorflow (it's OK if you don't know)?